pkg/ddl/index.go

```diff
@@ -2549,7 +2557,13 @@ func getNextPartitionInfo(reorg *reorgInfo, t table.PartitionedTable, currPhysic
 		if len(pi.AddingDefinitions) == 0 {
 			// case 1
 			// Simply AddIndex, without any partitions added or dropped!
-			pid, err = findNextPartitionID(currPhysicalTableID, pi.Definitions)
+			if reorg.mergingTmpIdx && currPhysicalTableID == t.Meta().ID {
+				// If the current Physical id is the table id,
+				// the next Physical id should be the first partition id.
+				pid = pi.Definitions[0].ID
```

Comment:
Should we loop through `reorg.elements` and match against `t.indices` to find the corresponding index (with the correct ID and state) and if every index in the elements are Global, set `pid = 0` to avoid searching for local temporary indexes?